integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/optimizer/CarbonSecondaryIndexOptimizer.scala

```diff
@@ -693,10 +693,10 @@ class CarbonSecondaryIndexOptimizer(sparkSession: SparkSession) {
       case sort@Sort(order, global, plan) =>
         addProjection = true
         (sort, true)
-      case filter@Filter(condition, logicalRelation@MatchIndexableRelation(indexableRelation))
+      case filter@Filter(condition, logicalRelation@MatchIndexTableRelation(indexTableRelation))
```

Comment:
rename is changing meaning. And actual code is to match CarbonDatasourceHadoopRelation

Reply:
Reverted the change.